extensions/answer_summarizers/models.py

```diff
@@ -203,36 +209,21 @@ class FrequencyCommonlySubmittedElements(BaseCalculation):
     """
 
     def calculate_from_state_answers_dict(self, state_answers_dict):
-        """Computes the number of occurrences of each element across
-        all given answers, keeping only the top 10 elements. Returns a
-        list of dicts; each dict has keys 'element' and 'frequency'.
+        """Computes the number of occurrences of each element across all given
+        answers, keeping only the top 10 elements. Returns a list of dicts; each
+        dict has keys 'answer' and 'frequency'.
 
         This method is run from within the context of a MapReduce job.
         """
-        answer_values = [
-            answer_dict['answer']
-            for answer_dict in state_answers_dict['submitted_answer_list']]
-
-        list_of_all_elements = []
-        for set_value in answer_values:
-            list_of_all_elements += set_value
-
-        elements_as_list_of_pairs = sorted(
-            collections.Counter(list_of_all_elements).items(),
-            key=lambda x: x[1],
-            reverse=True)
-        # Keep only top 10 elements
-        if len(elements_as_list_of_pairs) > 10:
-            elements_as_list_of_pairs = elements_as_list_of_pairs[:10]
-
-        calculation_output = []
-        for item in elements_as_list_of_pairs:
-            # Save element with key 'answer' so it gets displayed correctly
-            # by FrequencyTable visualization.
-            calculation_output.append({
-                'answer': item[0],
-                'frequency': item[1],
-            })
+        hashed_element_frequencies = collections.Counter()
+        for answer_dict in state_answers_dict['submitted_answer_list']:
+            hashed_element_frequencies.update(
+                _HashedValue(element) for element in answer_dict['answer'])
```

Comment:
Just a thought, I'm not sure why we're introducing the term "element" here (though I realize this was done before). Would it make better sense to just call it "answer" throughout?

Reply:
Done, went with "individual answer".